apps/stable_diffusion/src/models/model_wrappers.py

```diff
@@ -86,15 +156,22 @@ def __init__(
         generate_vmfb: bool = True,
         is_inpaint: bool = False,
         is_upscaler: bool = False,
+        is_sdxl: bool = False,
         use_stencil: str = None,
         use_lora: str = "",
         use_quantize: str = None,
         return_mlir: bool = False,
     ):
         self.check_params(max_len, width, height)
         self.max_len = max_len
-        self.height = height // 8
-        self.width = width // 8
+        self.is_sdxl = is_sdxl
+        self.height = height
+        self.width = width
+        if is_sdxl:
+            # We need to scale down the height/width by vae_scale_factor, which
+            # happens to be 8 in this case.
+            self.height = height // 8
+            self.width = width // 8
```

Comment:
This seems unintentional. The original SD needs `height // 8` and `width //  8`  and this changes the default while assigning the old behavior to SDXL.

If I try SD1.4 from scratch on this commit, I get a 4TB CPU allocation error. If I add `not` to `if is_sdxl` on line 170, the problem seems fixed for SD1.5/2.1 but now the VAE issue starts to make more sense...

I'll update after a bit more debugging.

Reply:
Oh, yeah I think it might be a good idea to just revert this one change entirely.... unless i'm mistaken

Reply:
Perhaps following would help clear why `// 8` (`vae_scale_factor`) needs to be done for SDXL :-

1. User only mentions the final image size to be generated, be it SD1.X/2.1/XL.
2. In case of SDXL we currently only deal with 1024x1024 image generation.
3. But the inputs to UNET/VAE for a 1024x1024 image needs to be 128x128. And this is nothing but `1024 // 8`.
4. This is something I observed in the normal Python execution by printing the input sizes expected by individual models right before their "forward" method is invoked.
5. Since the inputs to each model is worked out currently via `base_model.json` and for SD1.X/2.1 it substitutes the final height/width - but for SDXL it needs to be the scaled down version (as seen in point 3 above).
6. So, I added that and guarded it with `if is_sdxl` flag.

Reply:
Before this commit, the SD1.X / 2.1 models already do `height // 8` and `width // 8` here. This commit changes the behavior for those models at this point to `height` and `width` (introduces the CPUAllocator issue) and then defines a special case for XL with the old default behavior. Unless we are reworking the pipeline for SD1.X/2.1 let's keep the non-SDXL path as it was beforehand, to reduce breakages